rust/lance/src/io/commit/conflict_resolver.rs

```diff
@@ -827,13 +1002,152 @@ impl<'a> TransactionRebase<'a> {
                 | Operation::Restore { .. }
                 | Operation::ReserveFragments { .. }
                 | Operation::Update { .. }
-                | Operation::Project { .. } => Ok(()),
+                | Operation::Project { .. }
+                | Operation::UpdateMemWalState { .. } => Ok(()),
             }
         } else {
             Err(wrong_operation_err(&self.transaction.operation))
         }
     }
 
+    fn check_update_mem_wal_state_txn(
+        &mut self,
+        other_transaction: &Transaction,
+        other_version: u64,
+    ) -> Result<()> {
+        if let Operation::UpdateMemWalState {
+            added,
+            updated,
+            removed,
+            ..
+        } = &self.transaction.operation
+        {
+            match &other_transaction.operation {
+                Operation::UpdateMemWalState {
+                    added: committed_added,
+                    updated: committed_updated,
+                    removed: committed_removed,
+                } => {
+                    // 1. if the current or last committed job is trimming flushed MemWALs,
+                    // it is compatible with any other UpdateMemWalState commits
+                    if (committed_added.is_empty()
+                        && committed_updated.is_empty()
+                        && !committed_removed.is_empty())
+                        || (added.is_empty() && updated.is_empty() && !removed.is_empty())
```

Comment:
Why check `!committed_removed.is_empty()` and `!removed.is_empty()`?  I agree they shouldn't be empty if the other two are empty but does it matter?